eth/backend.go

```diff
@@ -299,16 +274,39 @@ func New(stack *node.Node, config *ethconfig.Config) (*Ethereum, error) {
 	}); err != nil {
 		return nil, err
 	}
-	if eth.votePool != nil {
-		eth.handler.votepool = eth.votePool
+
+	eth.miner = miner.New(eth, &config.Miner, chainConfig, eth.EventMux(), eth.engine, eth.isLocalBlock)
+	eth.miner.SetExtra(makeExtraData(config.Miner.ExtraData))
+
+	// Create voteManager instance
+	if posa, ok := eth.engine.(consensus.PoSA); ok {
```

Comment:
can we just add a check here, if mining is not enabled, no need to create: VotePool, VoteManager?

Reply:
no 
validator can start without the `mine` flag
but start mining later through geth console